tests/mpi/periodic_neighbor_01.cc

```diff
@@ -0,0 +1,371 @@
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       -------------------------------------------------
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       |           |           |           |           |
+ *       -------------------------------------------------
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           -------------------------           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       |           |     |     |     |     |           |
+ *       -------------------------------------------------
+ *                    Bottom periodic boundary
+ *
+ *
+ * =====================
+ * [ Passing criteria: ]
+ * =====================
+ *
+ * The test will be passed if the main three assertions are passed.
+ */
+
+#include "../tests.h"
+#include <deal.II/distributed/grid_refinement.h>
+#include <deal.II/distributed/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_boundary_lib.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/lac/vector.h>
+
+#include <fstream>
+#include <string>
+
+
+using namespace dealii;
+
+template <int dim>
+struct periodicity_tests
+{
+  typedef TriaIterator<CellAccessor<dim>> cell_iterator;
+  typedef typename Triangulation<dim>::active_cell_iterator active_cell_iterator;
+  periodicity_tests();
+  ~periodicity_tests();
+
+  unsigned refn_cycle;
+  MPI_Comm mpi_comm;
+  int comm_rank, comm_size;
+  parallel::distributed::Triangulation<dim> the_grid;
+
+  void refine_grid(const unsigned &);
+  void write_grid();
+  void check_periodicity();
+};
+
+template <int dim>
+periodicity_tests<dim>::periodicity_tests()
+  : refn_cycle(0), mpi_comm(MPI_COMM_WORLD), the_grid(mpi_comm)
+{
+  Assert(dim == 2 || dim == 3, ExcMessage("Only implemented for the 2D and 3D case!"));
+  comm_rank = Utilities::MPI::this_mpi_process(mpi_comm);
+  comm_size = Utilities::MPI::n_mpi_processes(mpi_comm);
+  std::vector<unsigned> repeats(dim, 2);
+  Point<dim> p1, p2, periodic_transfer;
+  if (dim == 2)
+    {
+      p2 = { 16., 16. };
```

Comment:
Line 134. It's not a C-style array initialization, but in C++11 is mapped to a copy constructor. This doesn't work in C++98, though. You'll have to write this as
```
  p2[0] = 16.;
  p2[1] = 16.;
```
or
```
  p2 = Point<dim>(16,16);
```
Same in line 139.